pkg/provider/aws/auth/auth.go

```diff
@@ -382,7 +382,7 @@ func getAWSSession(config *aws.Config, enableCache bool, name, kind, namespace,
 	}
 
 	if enableCache {
-		sessionCache.Add(resourceVersion, key, sess)
+		sessionCache.Add(resourceVersion, key, sess.Copy())
```

Comment:
this fixes a nasty bug. 
Before, this session is stored as a pointer (!) and the session credentials are manipulated [from here](https://github.com/external-secrets/external-secrets/blob/ff4342d943f10aaa8b7459632e0974265b173cbe/pkg/provider/aws/auth/auth.go#L152-L154) - effectively changing the value in cache.

As a result, the aws session stored in cache has credentials to the already assumed role.
When this session is retrieved from cache, eso will try to assume the role that is already there. This is not a problem, but in order to get this working the IAM Role required additional permissions (`sts:AssumeRole`). 

With this fix the session is stored with its initial credentials and the assumeRole operation does not mutate the session.